hypothesis-python/src/hypothesis/strategies/_internal/strategies.py

```diff
@@ -106,115 +115,112 @@ def recursive_property(name, default):
-            def recur_inner(other):
-                try:
-                    return forced_value(other)
-                except AttributeError:
-                    pass
-                listeners[other].add(strat)
-                result = mapping.get(other, sentinel)
-                if result is sentinel:
-                    needs_update.add(other)
-                    mapping[other] = default
-                    return default
-                return result
-
-            return recur_inner
-
-        count = 0
-        seen = set()
-        while needs_update:
-            count += 1
-            # If we seem to be taking a really long time to stabilize we
-            # start tracking seen values to attempt to detect an infinite
-            # loop. This should be impossible, and most code will never
-            # hit the count, but having an assertion for it means that
-            # testing is easier to debug and we don't just have a hung
-            # test.
-            # Note: This is actually covered, by test_very_deep_deferral
-            # in tests/cover/test_deferred_strategies.py. Unfortunately it
-            # runs into a coverage bug. See
-            # https://github.com/nedbat/coveragepy/issues/605
-            # for details.
-            if count > 50:  # pragma: no cover
-                key = frozenset(mapping.items())
-                assert key not in seen, (key, name)
-                seen.add(key)
-            to_update = needs_update
-            needs_update = set()
-            for strat in to_update:
-                new_value = getattr(strat, calculation)(recur2(strat))
-                if new_value != mapping[strat]:
-                    needs_update.update(listeners[strat])
-                    mapping[strat] = new_value
-
-        # We now have a complete and accurate calculation of the
-        # property values for everything we have seen in the course of
-        # running this calculation. We simultaneously update all of
-        # them (not just the strategy we started out with).
-        for k, v in mapping.items():
-            setattr(k, cache_key, v)
-        return getattr(self, cache_key)
-
-    accept.__name__ = name
-    return property(accept)
+        return recur_inner
+
+    count = 0
+    seen = set()
+    while needs_update:
+        count += 1
+        # If we seem to be taking a really long time to stabilize we
+        # start tracking seen values to attempt to detect an infinite
+        # loop. This should be impossible, and most code will never
+        # hit the count, but having an assertion for it means that
+        # testing is easier to debug and we don't just have a hung
+        # test.
+        # Note: This is actually covered, by test_very_deep_deferral
+        # in tests/cover/test_deferred_strategies.py. Unfortunately it
+        # runs into a coverage bug. See
+        # https://github.com/nedbat/coveragepy/issues/605
+        # for details.
```

Comment:
A check for an impossible case, which nonetheless has a test, except also a nocover pragma, due to an intractable upstream bug?  Classic Hypothesis experience, nothing to see here...